std/demangle.d

```diff
@@ -18,72 +18,73 @@
  */
 module std.demangle;
 
-/+
-private class MangleException : Exception
+/**
+Demangle D mangled names.
+
+Params:
+    name = the mangled name
+Returns:
+    A `string`. If it is not a D mangled name, it returns its argument name.
+ */
+string demangle(string name) @safe pure nothrow
 {
-    this()
-    {
-        super("MangleException");
-    }
+    import core.demangle : demangle;
+    auto ret = demangle(name);
+    return ret.idup;
 }
-+/
 
-/*****************************
- * Demangle D mangled names.
- *
- * If it is not a D mangled name, it returns its argument name.
- * Example:
- *        This program reads standard in and writes it to standard out,
- *        pretty-printing any found D mangled names.
--------------------
-import core.stdc.stdio : stdin;
-import std.stdio;
-import std.ascii;
-import std.demangle;
-
-void test(int x, float y) { }
+///
+@safe pure unittest
+{
+    // int in module a
+    assert(demangle("_D1a1bi") == "int a.b");
+    // char array in module test
+    assert(demangle("_D4test3fooAa") == "char[] test.foo");
+}
 
-int main()
+/**
+This program reads standard in and writes it to standard out,
+pretty-printing any found D mangled names.
```

Comment:
I'm not sure this is a good example for the documentation:
* it's style is very much C/D1
* it doesn't handle UTF8 correctly
* when running the example in the browser, there is no output

Reply:
An "D2-like" example:

```d
@safe unittest
{
      import std.ascii, std.algorithm, std.conv, std.demangle, std.functional, std.stdio;
    "aaa  ...  _D9main4testFifZv"
        .chunkBy!(a => std.ascii.isAlphaNum(a) || a == '_')
        .map!(a => a[1].pipe!(to!string, demangle))
        .joiner
        .writeln;
}
```

"D2-like", because demangle should accept a range and not only string to avoid the `to!string` allocation.

If we still want to support stdin/stdout, that would be sth. like:

```d
void main()
{
    import std.ascii, std.algorithm, std.conv, std.demangle, std.functional, std.stdio;
    stdin.byLineCopy
        .map!(l => l.chunkBy!(a => std.ascii.isAlphaNum(a) || a == '_').map!(a => a[1].pipe!(to!string, demangle)).joiner)
        .copy(stdout.lockingTextWriter);
}
```